src/Adapter/Product/Combination/Repository/CombinationRepository.php

```diff
@@ -143,14 +144,72 @@ public function partialUpdate(Combination $combination, array $updatableProperti
     }
 
     /**
-     * @param Combination $combination
+     * @param CombinationId $combinationId
      * @param int $errorCode
      *
      * @throws CoreException
      */
-    public function delete(Combination $combination, int $errorCode = 0): void
+    public function delete(CombinationId $combinationId, int $errorCode = 0): void
+    {
+        $this->deleteObjectModel($this->get($combinationId), CannotDeleteCombinationException::class, $errorCode);
+    }
+
+    /**
+     * @param ProductId $productId
+     *
+     * @throws CannotDeleteCombinationException
+     */
+    public function deleteByProductId(ProductId $productId): void
+    {
+        $combinationIds = $this->getCombinationIdsByProductId($productId);
+
+        $this->bulkDelete($combinationIds);
+    }
+
+    /**
+     * @param array $combinationIds
+     */
+    public function bulkDelete(array $combinationIds): void
```

Comment:
Initially I performed a bulk delete with SQL only, but we need to use the ObjectModel because some hooks are triggered And without them some modules (and their tables) are not correctly cleaned (like index table from faceted search) which causes bugs afterwards

Reply:
The limitation here will be when the product has hundreds or thousands of combinations